### PR TITLE
feat(docdb): implement audit and profiler logs

### LINKED ISSUE #17478: (aws_docdb): support audit logging
### Description

In order to use data stores while remaining HIPAA compliant, we need to log an audit trail. DocDB has support for this https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing but it doesn't look like there's any way to enable this via CDK. We had a similar problem with RDS and solved it using a CustomResource lambda. Is there something like this for DocDB?

### Use Case

HIPAA compliance

### Proposed Solution

new aws_docdb.DatabaseCluster(this, 'Name', { audit: true, ... });

When this is enabled, it would automatically create a customer Parameter Group to match the DB with `audit_logs: 'enabled'` and then it would trigger a custom resource that would run the equivalent of

```
aws docdb modify-db-cluster \
   --db-cluster-identifier sample-cluster \
   --cloudwatch-logs-export-configuration '{"EnableLogTypes":["audit"]}'
```

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by markussiebert**
closes #17478 

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-docdb/lib/props.ts [hunk 1]
```diff
@@ -92,3 +92,23 @@ export interface RotationMultiUserOptions {
    */
   readonly automaticallyAfter?: Duration;
 }
+
+/**
+ * Properties for the CloudwatchLogsExports
+ */
+
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/props.ts:99]: No need for this empty line.

File: packages/@aws-cdk/aws-docdb/lib/props.ts [hunk 2]
```diff
@@ -92,3 +92,23 @@ export interface RotationMultiUserOptions {
...
+export interface CloudwatchLogsExportsProps {
+  /**
+   * Should sending profiler logs to CloudWatch Logs should be enabled?
+   * You have to configure the profiler additionally
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling
+   * @default false
+   */
+  readonly profiler?: boolean;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/props.ts:107]: And here, we need an empty line after `readonly profiler?: boolean;` 🙂.

File: packages/@aws-cdk/aws-docdb/lib/props.ts [hunk 3]
```diff
@@ -92,3 +92,23 @@ export interface RotationMultiUserOptions {
...
+ * Properties for the CloudwatchLogsExports
+ */
+
+export interface CloudwatchLogsExportsProps {
+  /**
+   * Should sending profiler logs to CloudWatch Logs should be enabled?
+   * You have to configure the profiler additionally
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling
+   * @default false
+   */
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/props.ts:106]:
```suggestion
  /**
   * Whether the profiler logs should be exported to CloudWatch.
   *
   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling
   * @default false
   */
```

File: packages/@aws-cdk/aws-docdb/lib/props.ts [hunk 4]
```diff
@@ -92,3 +92,23 @@ export interface RotationMultiUserOptions {
...
+  readonly profiler?: boolean;
+  /**
+   * Should sending audit logs to CloudWatch Logs should be enabled?
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing
+   * @default false
+   */
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/props.ts:112]:
```suggestion
  /**
   * Whether the audit logs should be exported to CloudWatch.
   *
   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing
   * @default false
   */
```

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 5]
```diff
@@ -146,6 +147,14 @@ export interface DatabaseClusterProps {
...
   readonly deletionProtection?: boolean;
+
+  /**
+   * The configuration of log types that can be enabled for exporting to Amazon CloudWatch Logs.
+   * You can enable audit logs or profiler logs.
+   *
+   * @default {}
+   */
+  readonly cloudwatchLogsExports?: CloudwatchLogsExportsProps;
 }
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/cluster.ts:158]: Actually, let's get rid of the `CloudwatchLogsExportsProps` interface, and just have 2 separate properties in `DatabaseClusterProps`, `exportAuditLogsToCloudWatch?: boolean` and `exportProfileLogsToCloudWatch?: boolean`.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @markussiebert, looks great! A few minor comments.

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 6]
```diff
@@ -146,6 +147,14 @@ export interface DatabaseClusterProps {
...
    */
   readonly deletionProtection?: boolean;
+
+  /**
+   * The configuration of log types that can be enabled for exporting to Amazon CloudWatch Logs.
+   * You can enable audit logs or profiler logs.
+   *
+   * @default {}
+   */
+  readonly cloudwatchLogsExports?: CloudwatchLogsExportsProps;
```
[the remark this answers is not in the document]
Comment (markussiebert) [reply packages/@aws-cdk/aws-docdb/lib/cluster.ts:157]: Good point... AWS says "EnableCloudwatchLogsExports" in their cloudformation documentation

image

[the remark this answers is not in the document]
Comment (markussiebert) [reply packages/@aws-cdk/aws-docdb/lib/cluster.ts:157]: I sticked to this

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts (hunk 5 above)
Comment (markussiebert) [reply packages/@aws-cdk/aws-docdb/lib/cluster.ts:158]: it's ok

Comment (markussiebert) [reply packages/@aws-cdk/aws-docdb/lib/cluster.ts:166]: changed this

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 7]
```diff
@@ -8,7 +8,8 @@ import { DatabaseSecret } from './database-secret';
 import { CfnDBCluster, CfnDBInstance, CfnDBSubnetGroup } from './docdb.generated';
 import { Endpoint } from './endpoint';
 import { IClusterParameterGroup } from './parameter-group';
-import { BackupProps, Login, RotationMultiUserOptions } from './props';
+import { BackupProps, CloudwatchLogsExportsProps, Login, RotationMultiUserOptions } from './props';
+
```
[the remark this answers is not in the document]
Comment (markussiebert) [reply packages/@aws-cdk/aws-docdb/lib/cluster.ts:12]: 👍

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (markussiebert) [reply packages/@aws-cdk/aws-docdb/lib/cluster.ts]: @skinny85 what's your oppinion?

File: packages/@aws-cdk/aws-docdb/README.md [hunk 8]
```diff
@@ -115,3 +115,22 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
+
+Sending audit or profiler needs to be configured in two places, 
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and [profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
+2. Enable the corresponding option in the cloudwatchLogsExports properties
+
+```ts
+const cluster = new DatabaseCluster(this, 'Database', {
+    ...
+    cloudwatchLogsExports: {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:129]: These docs are now out of date.

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 9]
```diff
@@ -146,6 +146,22 @@ export interface DatabaseClusterProps {
...
    */
   readonly deletionProtection?: boolean;
+
+  /**
+   * Should sending profiler logs to CloudWatch Logs should be enabled?
+   * You have to configure the profiler additionally in the parameter group.
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling
+   * @default false
+   */
+  readonly exportProfilerLogsToCloudWatch?: boolean;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/cluster.ts:156]: Please incorporate this comment: https://github.com/aws/aws-cdk/pull/17570#discussion_r752788270.

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 10]
```diff
@@ -146,6 +146,22 @@ export interface DatabaseClusterProps {
...
+
+  /**
+   * Should sending audit logs to CloudWatch Logs should be enabled?
+   * You have to configure the audit additionally in the parameter group.
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing
+   * @default false
+   */
+  readonly exportAuditLogsToCloudWatch?: boolean;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/cluster.ts:164]: Please incorporate this comment: https://github.com/aws/aws-cdk/pull/17570#discussion_r752788459.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks great @markussiebert! Only minor documentations changes left.

File: packages/@aws-cdk/aws-docdb/README.md [hunk 11]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and [profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
+2. Enable the corresponding option in the cloudwatchLogsExports properties
+
+```ts
+const cluster = new DatabaseCluster(this, 'Database', {
+    ...
+    exportProfilerLogsToCloudWatch: true // Enable sending profiler logs
+    exportAuditLogsToCloudWatch: true // Enable sending audit logs
+    }
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:132]:
```suggestion
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 12]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
+Sending audit or profiler needs to be configured in two places, 
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and [profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
+2. Enable the corresponding option in the cloudwatchLogsExports properties
+
+```ts
+const cluster = new DatabaseCluster(this, 'Database', {
+    ...
+    exportProfilerLogsToCloudWatch: true // Enable sending profiler logs
+    exportAuditLogsToCloudWatch: true // Enable sending audit logs
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:131]:
```suggestion
  ...,
  exportProfilerLogsToCloudWatch: true // Enable sending profiler logs
  exportAuditLogsToCloudWatch: true // Enable sending audit logs
```

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 13]
```diff
@@ -146,6 +146,24 @@ export interface DatabaseClusterProps {
...
+   */
+  readonly exportProfilerLogsToCloudWatch?: boolean;
+
+  /**
+   * Should sending audit logs to CloudWatch Logs be enabled?
+   * You have to configure the audit additionally in the parameter group.
+   *
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing
+   * @default false
+   */
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/cluster.ts:165]:
```suggestion
  /**
   * Whether the audit logs should be exported to CloudWatch.
   * Note that you also have to configure the audit log export in the Cluster's Parameter Group.
   *
   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing
   * @default false
   */
```

File: packages/@aws-cdk/aws-docdb/lib/cluster.ts [hunk 14]
```diff
@@ -146,6 +146,24 @@ export interface DatabaseClusterProps {
...
    */
   readonly deletionProtection?: boolean;
+
+  /**
+   * Should sending profiler logs to CloudWatch Logs be enabled?
+   * You have to configure the profiler additionally in the parameter group.
+   *
+   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling
+   * @default false
+   */
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/lib/cluster.ts:156]:
```suggestion
  /**
   * Whether the profiler logs should be exported to CloudWatch.
   * Note that you also have to configure the profiler log export in the Cluster's Parameter Group.
   *
   * @see https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling
   * @default false
   */
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 15]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
 The rotation will start as soon as this user exists.
 
 See also [@aws-cdk/aws-secretsmanager](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-secretsmanager/README.md) for credentials rotation of existing clusters.
+
+## Audit and profiler Logs
+
+Sending audit or profiler needs to be configured in two places, 
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and [profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
+2. Enable the corresponding option in the cloudwatchLogsExports properties
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:125]:
```suggestion
2. Enable the corresponding option in the cloudwatchLogsExports properties.
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 16]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
 The rotation will start as soon as this user exists.
 
 See also [@aws-cdk/aws-secretsmanager](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-secretsmanager/README.md) for credentials rotation of existing clusters.
+
+## Audit and profiler Logs
+
+Sending audit or profiler needs to be configured in two places, 
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and [profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:124]:
```suggestion
1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and
[profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 17]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
 The rotation will start as soon as this user exists.
 
 See also [@aws-cdk/aws-secretsmanager](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-secretsmanager/README.md) for credentials rotation of existing clusters.
+
+## Audit and profiler Logs
+
+Sending audit or profiler needs to be configured in two places,
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:122]:
```suggestion
Sending audit or profiler needs to be configured in two places:
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 18]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
 
 See also [@aws-cdk/aws-secretsmanager](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-secretsmanager/README.md) for credentials rotation of existing clusters.
+
+## Audit and profiler Logs
+
+Sending audit or profiler needs to be configured in two places:
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and
+[profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
+2. Enable the corresponding option in the cloudwatchLogsExports properties.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:126]:
```suggestion
2. Enable the corresponding option(s) when creating the `DatabaseCluster`:
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 19]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
+
+```ts
+const cluster = new DatabaseCluster(this, 'Database', {
+  ...,
+  exportProfilerLogsToCloudWatch: true // Enable sending profiler logs
+  exportAuditLogsToCloudWatch: true // Enable sending audit logs
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:132]:
```suggestion
  exportProfilerLogsToCloudWatch: true // enable sending profiler logs
  exportAuditLogsToCloudWatch: true // enable sending audit logs
```

Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md]:
```suggestion
  exportProfilerLogsToCloudWatch: true, // Enable sending profiler logs
  exportAuditLogsToCloudWatch: true, // Enable sending audit logs
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 20]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
+
+1. Check / create the needed options in your ParameterGroup for [audit](https://docs.aws.amazon.com/documentdb/latest/developerguide/event-auditing.html#event-auditing-enabling-auditing) and
+[profiler](https://docs.aws.amazon.com/documentdb/latest/developerguide/profiling.html#profiling.enable-profiling) logs.
+2. Enable the corresponding option(s) when creating the `DatabaseCluster`:
+
+```ts
+const cluster = new DatabaseCluster(this, 'Database', {
+  ...,
+  exportProfilerLogsToCloudWatch: true // enable sending profiler logs
+  exportAuditLogsToCloudWatch: true // enable sending audit logs
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:132]:
```suggestion
  exportProfilerLogsToCloudWatch: true, // Enable sending profiler logs
  exportAuditLogsToCloudWatch: true, // Enable sending audit logs
```

File: packages/@aws-cdk/aws-docdb/README.md [hunk 21]
```diff
@@ -116,3 +116,21 @@ cluster.addRotationMultiUser('MyUser', { // Add rotation using the multi user sc
...
+});
+```
+
+**Note** You have to do both steps, to activate logging.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-docdb/README.md:136]:
```suggestion
```

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @markussiebert!

**Pull request merged**